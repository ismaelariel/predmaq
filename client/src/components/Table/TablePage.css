.div_grid_conatiner {
    width: 100%;
    height: 50%;
    font-size: small;
    position: relative;
    overflow-y: scroll;
}