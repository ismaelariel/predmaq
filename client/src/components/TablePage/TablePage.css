.div_grid_conatiner {
    width: 100%;
    height: 50%;
    font-size: small;
    position: relative;
    overflow-y: scroll;
}

table {
    width: 100%;
    height: auto;
}

td, th {
    flex: 1;
    margin: 0;
    padding: 8px 0;
    text-align: center;
    color: #0000ff;
    border-bottom: 1px solid #0000ff;
}