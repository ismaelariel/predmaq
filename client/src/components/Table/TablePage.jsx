import React from "react";

import Table from "react-bootstrap/Table";

import "./TablePage.css";

const TablePage = ({machine}) => {
    return(
        <div className="div_grid_conatiner">
            <Table striped className="table">
                <thead>
                    <tr>
                        <th>AirTemperature</th>
                        <th>FailureType</th>
                        <th>ProcessTemperature</th>
                        <th>ProductID</th>
                        <th>RotationalSpeed</th>
                        <th>Target</th>
                        <th>ToolWear</th>
                        <th>Torque</th>
                        <th>Type</th>
                        <th>UDI</th>
                    </tr>
                </thead>
                <tbody>
                    {
                        machine.map((data) => {
                            return(
                                <tr>
                                    <td>{data.AirTemperature}</td>
                                    <td>{data.FailureType}</td>
                                    <td>{data.ProcessTemperature}</td>
                                    <td>{data.ProductID}</td>
                                    <td>{data.RotationalSpeed}</td>
                                    <td>{data.Target}</td>
                                    <td>{data.ToolWear}</td>
                                    <td>{data.Torque}</td>
                                    <td>{data.Type}</td>
                                    <td>{data.UDI}</td>
                                </tr>
                            )
                        })
                    }
                </tbody>
            </Table>
        </div>
    );
};

export default TablePage;