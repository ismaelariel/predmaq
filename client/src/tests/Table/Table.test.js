import React from "react";
import renderer from "react-test-renderer";

import TablePage from "../../components/Table/TablePage";

const machine = [{
    _id: '6443e224931711288579065c',
    UDI: 1,
    ProductID: 'M14860',
    Type: 'M',
    AirTemperature: 298.1,
    ProcessTemperature: 308.6,
    RotationalSpeed: 1551,
    Torque: 42.8,
    ToolWear: 0,
    Target: 0,
    FailureType: 'No Failure'
}];

describe('TablePage', () => {
    it('render table page component', () => {
        const component = renderer.create(
            <TablePage  predict={machine} />,
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});