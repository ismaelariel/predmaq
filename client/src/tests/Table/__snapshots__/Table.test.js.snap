// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePage render table page component 1`] = `
<div
  className="div_grid_conatiner"
>
  <table
    className="table table table-striped"
  >
    <thead>
      <tr>
        <th>
          AirTemperature
        </th>
        <th>
          FailureType
        </th>
        <th>
          ProcessTemperature
        </th>
        <th>
          ProductID
        </th>
        <th>
          RotationalSpeed
        </th>
        <th>
          Target
        </th>
        <th>
          ToolWear
        </th>
        <th>
          Torque
        </th>
        <th>
          Type
        </th>
        <th>
          UDI
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          298.1
        </td>
        <td>
          No Failure
        </td>
        <td>
          308.6
        </td>
        <td>
          M14860
        </td>
        <td>
          1551
        </td>
        <td>
          0
        </td>
        <td>
          0
        </td>
        <td>
          42.8
        </td>
        <td>
          M
        </td>
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
